### PR TITLE
Improve test and load

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
* Add command in group `Functional Tests` to run tests with option `--golden_run`.
* Add setting parameter to use the user `SCP_PRIVKEY` key when loading/deleting app on device.
* Add setting parameter to set the default device.

### CODE REVIEW

File: src/taskProvider.ts [hunk 1]
```diff
@@ -338,8 +351,13 @@ export class TaskProvider implements vscode.TaskProvider {
   private appDeleteExec(): string {
     let exec = "";
     if (platform === "linux") {
+      let scpconfig = "";
+      const conf = vscode.workspace.getConfiguration("ledgerDevTools");
```
Comment (agrojean-ledger) [inline src/taskProvider.ts:355]: We get the conf in the constructor and the `generateTasks` method, you don't need to have a conf variable here. Cf. 
https://github.com/LedgerHQ/ledger-vscode-extension/blob/23d15494c22d29ab3e26d4cc05b63c754fab08f1/src/taskProvider.ts#L177

So you can have a shared variables between task builders.

File: src/taskProvider.ts [hunk 2]
```diff
@@ -338,8 +351,13 @@ export class TaskProvider implements vscode.TaskProvider {
...
+      if (conf.get<boolean>("userScpPrivateKey") === true) {
+        scpconfig = "-e SCP_PRIVKEY=${SCP_PRIVKEY}";
+      }
       // Linux
-      exec = `docker exec -it  ${this.containerName} bash -c 'export BOLOS_SDK=$(echo ${getSelectedSDK()}) && make delete'`;
+      exec = `docker exec -it ${scpconfig} ${this.containerName} bash -c 'export BOLOS_SDK=$(echo ${getSelectedSDK()}) && make delete'`;
     } else if (platform === "darwin") {
```
Comment (agrojean-ledger) [inline src/taskProvider.ts:361]: You cannot support other platforms (windows, macos) ?

File: src/taskProvider.ts [hunk 3]
```diff
@@ -393,6 +411,14 @@ export class TaskProvider implements vscode.TaskProvider {
     return exec;
   }
 
+  private functionalTestsGoldenRunExec(): string {
+    // Runs functional tests inside the docker container (with Qt display enabled).
```
Comment (agrojean-ledger) [inline src/taskProvider.ts:415]: Update the comment

File: src/taskProvider.ts [hunk 4]
```diff
@@ -107,6 +107,14 @@ export class TaskProvider implements vscode.TaskProvider {
       toolTip: "Run Python functional tests (with Qt display enabled)",
       enabled: true,
     },
+    {
+      group: "Functional Tests",
+      name: "Run tests with option 'golden_display'",
```
Comment (agrojean-ledger) [inline src/taskProvider.ts:112]: I would rather have something like `Generate golden tests snapshots` displayed to the user

File: src/taskProvider.ts [hunk 5]
```diff
@@ -107,6 +107,14 @@ export class TaskProvider implements vscode.TaskProvider {
...
+      builders: { ["Both"]: this.functionalTestsGoldenRunExec },
+      dependsOn: this.functionalTestsRequirementsExec,
+      toolTip: "Run Python functional tests while storing the generated png files",
```
Comment (agrojean-ledger) [inline src/taskProvider.ts:115]: "Run Python functional tests while storing the generated png files" 

I'd reword to 

"Run Python functional tests with --golden_run option to generate golden snapshots. They are used during tests runs to check what should be displayed on the device screen"

File: src/taskProvider.ts (hunk 1 above)
Comment (agrojean-ledger) [reply src/taskProvider.ts:355]: The generateTasks method is called each time the settings are changed by the user, so we are sure the tasks use the latest setting.

File: src/taskProvider.ts (hunk 5 above)
Comment (cedelavergne-ledger) [reply src/taskProvider.ts:115]: ok

File: src/taskProvider.ts (hunk 4 above)
Comment (cedelavergne-ledger) [reply src/taskProvider.ts:112]: ok

File: src/taskProvider.ts (hunk 3 above)
Comment (cedelavergne-ledger) [reply src/taskProvider.ts:415]: ok

File: src/taskProvider.ts (hunk 1 above)
Comment (cedelavergne-ledger) [reply src/taskProvider.ts:355]: Fixed!

File: src/taskProvider.ts (hunk 2 above)
Comment (cedelavergne-ledger) [reply src/taskProvider.ts:361]: Done

File: README.md [hunk 6]
```diff
@@ -50,9 +50,15 @@ This extension contributes the following settings:
 * `ledgerDevTools.additionalDepsPerApp`: Add dependencies for current app's functional tests (for instance 'apk add python3-protobuf').
 * `ledgerDevTools.keepContainerTerminal`: Indicates to keep the Terminal window opened after a successful Container Update.
 * `ledgerDevTools.containerUpdateRetries`: Set the max number of Container Update retries.
+* `ledgerDevTools.userScpPrivateKey`: Use the user `SCP_PRIVATE` key when loading/deleting app on device.
```
Comment (agrojean-ledger) [inline README.md:53]:
```suggestion
* `ledgerDevTools.userScpPrivateKey`: Use the host's `SCP_PRIVKEY` environment variable when loading/deleting app on device. Cf. https://developers.ledger.com/docs/embedded-app/pin-bypass/
```

File: package.json [hunk 7]
```diff
@@ -117,7 +117,13 @@
           "type": "number",
           "default": 5,
           "markdownDescription": "Max number of Container Update retries."
+        },
+        "ledgerDevTools.userScpPrivateKey": {
+          "type": "boolean",
+          "default": false,
+          "markdownDescription": "Use the user SCP_PRIVATE key when loading/deleting app on device."
```
Comment (agrojean-ledger) [inline package.json:124]:
```suggestion
          "markdownDescription": "Use the host's `SCP_PRIVKEY` environment variable when loading/deleting app on device. Cf. https://developers.ledger.com/docs/embedded-app/pin-bypass/"
```

File: src/taskProvider.ts [hunk 8]
```diff
@@ -319,36 +330,56 @@ export class TaskProvider implements vscode.TaskProvider {
...
+      if (this.scpConfig === true) {
+        keyconfig = "-e SCP_PRIVKEY=${SCP_PRIVKEY}";
+      }
       // Executes make load in the container to load the app on a physical device.
-      exec = `docker exec -it  ${this.containerName} bash -c 'export BOLOS_SDK=$(echo ${getSelectedSDK()}) && make load'`;
+      exec = `docker exec -it ${keyconfig} ${this.containerName} bash -c 'export BOLOS_SDK=$(echo ${getSelectedSDK()}) && make load'`;
     } else if (platform === "darwin") {
       // macOS
+      if (this.scpConfig === true) {
+        keyconfig = "--rootPrivateKey ${SCP_PRIVKEY}";
```
Comment (agrojean-ledger) [inline src/taskProvider.ts:344]: Did you try it ? Not sure there would be proper variable expansion :thinking:

Comment (cedelavergne-ledger) [reply src/taskProvider.ts:344]: I don't have a Mac to test (neither Windows). I just kept the same syntax as the other variables in the exec command, like `${this.appName}`

File: README.md (hunk 6 above)
Comment (cedelavergne-ledger) [reply README.md]: Done

File: package.json (hunk 7 above)
Comment (cedelavergne-ledger) [reply package.json]: done

File: src/taskProvider.ts (hunk 8 above)
Comment (agrojean-ledger) [reply src/taskProvider.ts:344]: `${<variable>}` is for expanding program variables when using ` ` ` quotes around a string. Not for expanding host's environment variables. We should test at least on one of the other platforms (either windows or mac - on a VM if needed) before releasing. This is painful but one of the goals of this extension is to be cross platform.

Comment (agrojean-ledger) [reply src/taskProvider.ts:344]: For Windows there are official virtual machine images available on Microsoft's website : https://developer.microsoft.com/en-us/windows/downloads/virtual-machines/ (I tried the vmware image and it works Ok on linux with vmware player)

Comment (cedelavergne-ledger) [reply src/taskProvider.ts:344]: Done. tested and it works also with Windows

Comment (cedelavergne-ledger) [reply src/taskProvider.ts]: Tested also OK on Mac
